script/releases/v0.1.0-rewardsv2/2-multisig.s.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import {MultisigCall, MultisigCallUtils, MultisigBuilder} from "zeus-templates/templates/MultisigBuilder.sol";
+import {ITimelock} from "zeus-templates/interfaces/ITimelock.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import {RewardsCoordinator} from "src/contracts/core/RewardsCoordinator.sol";
+import {EigenLabsUpgrade} from "../EigenLabsUpgrade.s.sol";
+
+/**
+ * Purpose: enqueue a multisig transaction which tells the ProxyAdmin to upgrade RewardsCoordinator. 
+ */
+contract Queue is MultisigBuilder {
+    using MultisigCallUtils for MultisigCall[];
+    using EigenLabsUpgrade for *;
+
+    MultisigCall[] private _executorCalls;
+    MultisigCall[] private _opsCalls;
+
+    function _queue() internal returns (MultisigCall[] memory) {
+        uint32 rewardsCoordinatorSplitBips = zUpdateUint32(string("REWARDS_COORDINATOR_DEFAULT_OPERATOR_SPLIT_BIPS"), uint32(1000));
+
+        _executorCalls.append({
+            to: this._proxyAdmin(),
+            data: abi.encodeWithSelector(
+                ProxyAdmin.upgradeAndCall.selector,
+                zDeployedProxy(type(RewardsCoordinator).name),
+                zDeployedImpl(type(RewardsCoordinator).name),
+                abi.encodeWithSelector(
+                    RewardsCoordinator.initialize.selector,
+                    this._executorMultisig(),
+                    this._pauserRegistry(),
+                    zUint64("REWARDS_COORDINATOR_INIT_PAUSED_STATUS"),
+                    zAddress("REWARDS_COORDINATOR_UPDATER"),
+                    zUint64("REWARDS_COORDINATOR_ACTIVATION_DELAY"),
+                    rewardsCoordinatorSplitBips
+                )
+            )
+        });
+
+        return _executorCalls;
+    }
+
+    function _execute() internal virtual override returns (MultisigCall[] memory) {
+        // get the queue data
+        MultisigCall[] memory calls = _queue();
+
+        address multiSendCallOnly = this._multiSendCallOnly();
+        address timelock = this._timelock();
+
+        // encode calls for executor
+        bytes memory executorCalldata = calls.makeExecutorCalldata(multiSendCallOnly, timelock);
+
+        address executorMultisig = this._executorMultisig();
+
+        // encode executor data for timelock
+        bytes memory timelockCalldata = abi.encodeWithSelector(
+            ITimelock.queueTransaction.selector, executorMultisig, 0, "", executorCalldata, type(uint256).max
+        );
+
+        _opsCalls.append(timelock, timelockCalldata);
+
+        // encode timelock data for ops multisig
+        return _opsCalls;
+    }
+
+    function testDeploy() virtual public {
+        _execute();
```

Comment:
@wadealexc @nadir-akhtar @0xrajath  - how/what do you envision us testing in these scripts for the multisig steps?

Reply:
I would like to mock approvals + timelock forwarding (if any) in these tests.

Reply:
Oh wait. The actual execute happens in Step 3? So then we have to test that it's been queued in the timelock here.